src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -396,16 +403,74 @@ def shared_prefix(candidate):
     shared_packages = [_f for _f in map(shared_prefix, all_packages) if _f]
     return '.'.join(max(shared_packages, key=len)) if shared_packages else package
 
-  @classmethod
-  def find_packages(cls, chroot, log=None):
+  @memoized_property
+  def nsutil_pex(self):
+    interpreter = self.context.products.get_data(PythonInterpreter)
+    chroot = os.path.join(self.workdir, 'nsutil', interpreter.version_string)
+    if not os.path.exists(chroot):
+      pex_info = PexInfo.default(interpreter=interpreter)
+      with safe_concurrent_creation(chroot) as scratch:
```

Comment:
Is "anonymous pex for code string" utility function worthy?

Reply:
Perhaps. I think https://github.com/pantsbuild/pants/issues/6401 will force the issue and require similar treatment of our generation of `setup.py`.